lib/resource-providers/vpc.ts

```diff
@@ -2,26 +2,29 @@ import { Tags } from 'aws-cdk-lib';
 import * as ec2 from 'aws-cdk-lib/aws-ec2';
 import { ISubnet, PrivateSubnet } from 'aws-cdk-lib/aws-ec2';
 import { ResourceContext, ResourceProvider } from "../spi";
+import * as eks from "aws-cdk-lib/aws-eks";
+import {Ipv6VpcProvider} from "./ipv6-vpc";
 
 /**
  * Interface for Mapping for fields such as Primary CIDR, Secondary CIDR, Secondary Subnet CIDR.
  */
 interface VpcProps {
-   primaryCidr: string, 
+   primaryCidr?: string,
    secondaryCidr?: string,
    secondarySubnetCidrs?: string[]
 }
 
 /**
  * VPC resource provider 
  */
-export class VpcProvider implements ResourceProvider<ec2.IVpc> {
+export class VpcProvider extends Ipv6VpcProvider {
```

Comment:
Do we really achieve much with inheritance here? Seems like delegation is cleaner option here. 

Ideally, we should have `CreateIpv6VpcProvider` and `CreateIpV4VpcProvider` and just delegate to one vs the other in the provide method if VPC needs to be created.